sdk/keyvault/keyvault-keys/test/tracing.test.ts

```diff
@@ -0,0 +1,245 @@
+  name?: string;
+  roots?: TestRecursiveGraph[];
+  children?: TestRecursiveGraph[];
+}
+
+/**
+ * Gets the names of all of the nodes of a SpanGraph from @azure/core-tracing
+ */
+function getNames(graph: TestRecursiveGraph): string[] {
+  const { name, roots, children = roots } = graph;
+  if (!children || !children!.length) {
+    return [];
+  }
+  const childrenNames = children.map(getNames).flat();
+  return [name!, ...childrenNames].filter((x) => x);
+}
+
+export type TracingTestFunc = (
+  this: MochaTestContext,
+  tracingOptions: OperationTracingOptions
+) => Promise<void>;
+
+interface TestTracingNodeNames {
+  (nodeNames: string[], test: TracingTestFunc): Test;
+  skip: (nodeNames: string[], test: TracingTestFunc) => Test;
+  only: (nodeNames: string[], test: TracingTestFunc) => Test;
+}
+
+function generateTracingTest(nodeNames: string[], test: TracingTestFunc): Mocha.Func {
+  return async function() {
+    const tracer = new TestTracer();
+    setTracer(tracer);
+    const rootSpan = tracer.startSpan("root");
+
+    await test.call(this, {
+      spanOptions: { parent: rootSpan.context() }
+    });
+
+    const graph = tracer.getSpanGraph(rootSpan.context().traceId);
+    const names = getNames(graph);
+    const uniqueNames = names.reduce(
+      (unique, next) => (unique.includes(next) ? unique : [...unique, next]),
+      ["root"]
+    );
+    assert.deepEqual(["root", ...nodeNames], uniqueNames);
+
+    rootSpan.end();
+    const activeSpans = tracer.getActiveSpans();
+    activeSpans.map((span) => console.log(`OPEN SPAN: ${span.name}`));
+    assert.equal(activeSpans.length, 0);
+  };
+}
+
+export const testTracingNodeNames: TestTracingNodeNames = function(
+  nodeNames: string[],
+  test: TracingTestFunc
+): Test {
+  return it(`traces the methods ${nodeNames.join(", ")}`, generateTracingTest(nodeNames, test));
+};
+
+testTracingNodeNames.skip = function(nodeNames: string[]): Test {
+  return it.skip(`Skipping tracing tests for the methods ${nodeNames.join(", ")}`);
+};
+
+testTracingNodeNames.only = function(nodeNames: string[], test: TracingTestFunc): Test {
+  return it.only(`traces the methods ${nodeNames.join(", ")}`, generateTracingTest(nodeNames, test));
+};
+
+describe("Keys client - tracing spans", () => {
+  const keyPrefix = `tracing${env.KEY_NAME || "KeyName"}`;
+  let keySuffix: string;
+  let client: KeyClient;
+  let testClient: TestClient;
+  let recorder: Recorder;
+
+  function getKeyName(context: MochaTestContext): string {
+    return testClient.formatName(`${keyPrefix}-${context!.test!.title}-${keySuffix}`);
+  }
+
+  beforeEach(async function() {
+    const authentication = await authenticate(this);
+    keySuffix = authentication.keySuffix;
+    client = authentication.client;
+    testClient = authentication.testClient;
+    recorder = authentication.recorder;
+  });
+
+  afterEach(async function() {
+    recorder.stop();
+  });
+
+  // The tests follow
+
+  testTracingNodeNames(
```

Comment:
what is the "name" of this test, according to mocha?

Reply:
In this case it says `traces the methods createKey, deleteKey...`, which is made by: https://github.com/Azure/azure-sdk-for-js/pull/8926/files/6597ad693ab6728e2aba3a3f786edb6a3155c15d#diff-f535315093fae96f7dabe5a1e99c08afR78